.github/workflows/go_build.yaml

```diff
@@ -68,4 +68,93 @@ jobs:
 
     - name: Spec Test Coverage
       run: |
-            go run tests/spectests/cmd/analyzer/analyzer.go
\ No newline at end of file
+            # Output the full report to console.
+            go run tests/spectests/cmd/analyzer/analyzer.go
+
+
+  go_spec_test_badge_update:
+    # change later to on merge: github.event.pull_request.merged
+    # Merged onto the default branch.
+    # later to enable for default branch, add: && github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+    if: github.event_name == 'pull_request'
```

Comment:
can we just use the [on directive](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on)?

```
on: [push, pull_request]
```

Would we want to only do this for main branch pushes? If so, check out https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches